igrill.py

```diff
@@ -177,8 +193,13 @@ def run(self):
                 while True:
                     temperature = device.read_temperature()
                     battery = device.read_battery()
-                    utils.publish(temperature, battery, self.mqtt_client, self.topic, device.name)
-                    logging.debug("Published temp: {} and battery: {} to topic {}/{}".format(temperature, battery, self.topic, device.name))
+                    heatingelement = device.read_heating_elements()
+                    if device.name == "Pulse 2000":
```

Comment:
No need to check, just pass the heatingelement variable to utils.publish no matter what it is.